src/bin/commands/containers.rs

```diff
@@ -0,0 +1,499 @@
+        list_volumes(self, &engine)
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct RemoveVolumes {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Force removal of volumes.
+    #[clap(short, long)]
+    pub force: bool,
+    /// Remove volumes. Default is a dry run.
+    #[clap(short, long)]
+    pub execute: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl RemoveVolumes {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        remove_volumes(self, &engine)
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct PruneVolumes {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Remove volumes. Default is a dry run.
+    #[clap(short, long)]
+    pub execute: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl PruneVolumes {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        prune_volumes(self, &engine)
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct CreateCrateVolume {
+    /// Triple for the target platform.
+    #[clap(long)]
+    pub target: String,
+    /// If cross is running inside a container.
+    #[clap(short, long)]
+    pub docker_in_docker: bool,
+    /// If we should copy the cargo registry to the volume.
+    #[clap(short, long)]
+    pub copy_registry: bool,
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl CreateCrateVolume {
+    pub fn run(self, engine: docker::Engine, channel: Option<&str>) -> cross::Result<()> {
+        create_crate_volume(self, &engine, channel)
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct RemoveCrateVolume {
+    /// Triple for the target platform.
+    #[clap(long)]
+    pub target: String,
+    /// If cross is running inside a container.
+    #[clap(short, long)]
+    pub docker_in_docker: bool,
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl RemoveCrateVolume {
+    pub fn run(self, engine: docker::Engine, channel: Option<&str>) -> cross::Result<()> {
+        remove_crate_volume(self, &engine, channel)
+    }
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Volumes {
+    /// List cross data volumes in local storage.
+    List(ListVolumes),
+    /// Remove cross data volumes in local storage.
+    Remove(RemoveVolumes),
+    /// Prune volumes not used by any container.
+    Prune(PruneVolumes),
+    /// Create a persistent data volume for the current crate.
+    CreateCrate(CreateCrateVolume),
+    /// Remove a persistent data volume for the current crate.
+    RemoveCrate(RemoveCrateVolume),
```

Comment:
these should have an alias I think

`create-crate` is a mouthful

simply doing `cross-util volumes create` and `cross-util volumes remove` is enough I think

Reply:
We already have `volumes remove` though and it has different functionality. We could do `list-all` and `remove-all`, but this is slightly different than the `images` API.

Reply:
Ok I've changed it to `list`, `remove-all` for containers and volumes, and then changed it to `create` and `remove` for persistent volumes, which now mention it's for the toolchain and not for the current crate.